### PR TITLE
To `polkadot-v0.9.40`

### PR DESCRIPTION
**Pull request by AurevoirXavier**
### Optional
- [ ] Companion of paritytech/cumulus#2180
- [ ] Companion of paritytech/cumulus#2290

### CODE REVIEW

**Comment by AurevoirXavier** [comment]
@boundless-forest Please help fix the eth rpc parts.

@jiguantong  Please help fix the start dev node parts (manual sealing).

**Comment by boundless-forest** [comment]
Command: `cargo build --release -p darwinia --all-features`  with the following error: @AurevoirXavier 


image

**Comment by jiguantong** [comment]
Dev node is ok
```
root@debian:~/dar-git/darwinia# ./target/release/darwinia --chain pangolin-dev --alice
2023-05-23 14:15:44 Darwinia    
2023-05-23 14:15:44 ✌️  version 6.3.0-c27fcdfcf62    
2023-05-23 14:15:44 ❤️  by Darwinia Network <hello@darwinia.network>, 2018-2023    
2023-05-23 14:15:44 📋 Chain specification: Pangolin2 Development    
2023-05-23 14:15:44 🏷  Node name: Alice    
2023-05-23 14:15:44 👤 Role: AUTHORITY    
2023-05-23 14:15:44 💾 Database: RocksDb at /root/.local/share/darwinia/chains/pangolin2-development/db/full    
2023-05-23 14:15:44 ⛓  Native runtime: Pangolin2-6300 (DarwiniaOfficialRust-0.tx0.au0)    
2023-05-23 14:15:45 [pallet::staking] assembling new collators for new session 0 at #0    
2023-05-23 14:15:45 [pallet::staking] assembling new collators for new session 1 at #0    
2023-05-23 14:15:45 Parachain id: Id(2105)    
2023-05-23 14:15:45 Parachain Account: 5Ec4AhNxga1JYLioRBNxfRnovheDELVbZTRSnKMgvSVPvNcN    
2023-05-23 14:15:45 Parachain genesis state: 0x000000000000000000000000000000000000000000000000000000000000000000c32d8524b53170e4ebc4c13069d3f330426655acb8be128d2d1e39e3335b53a003170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c11131400    
2023-05-23 14:15:45 Is collating: yes    
2023-05-23 14:15:45 [pallet::staking] assembling new collators for new session 0 at #0    
2023-05-23 14:15:45 [pallet::staking] assembling new collators for new session 1 at #0    
2023-05-23 14:15:46 🏷  Local node identity is: 12D3KooWS7arrV9axUQwfg2j2d3AVSQv6SYLhu4kxXFqHBcKD8U6    
2023-05-23 14:15:46 💻 Operating system: linux    
2023-05-23 14:15:46 💻 CPU architecture: x86_64    
2023-05-23 14:15:46 💻 Target environment: gnu    
2023-05-23 14:15:46 💻 CPU: 13th Gen Intel(R) Core(TM) i9-13900K    
2023-05-23 14:15:46 💻 CPU cores: 24    
2023-05-23 14:15:46 💻 Memory: 31840MB    
2023-05-23 14:15:46 💻 Kernel: 6.1.0-5-amd64    
2023-05-23 14:15:46 💻 Linux distribution: Debian GNU/Linux bookworm/sid    
2023-05-23 14:15:46 💻 Virtual machine: no    
2023-05-23 14:15:46 📦 Highest known block at #0    
2023-05-23 14:15:46 〽️ Prometheus exporter started at 127.0.0.1:9615    
2023-05-23 14:15:46 Running JSON-RPC HTTP server: addr=127.0.0.1:9933, allowed origins=["http://localhost:*", "http://127.0.0.1:*", "https://localhost:*", "https://127.0.0.1:*", "https://polkadot.js.org"]    
2023-05-23 14:15:46 Running JSON-RPC WS server: addr=127.0.0.1:9944, allowed origins=["http://localhost:*", "http://127.0.0.1:*", "https://localhost:*", "https://127.0.0.1:*", "https://polkadot.js.org"]    
2023-05-23 14:15:48 🙌 Starting consensus session on top of parent 0x48059233021f6d4bb957cc7349c335690d90760ac3433b8292fd2af5f96ce1fa    
2023-05-23 14:15:48 🎁 Prepared block for proposing at 1 (0 ms) [hash: 0xb4fbb7932aaa80a0b08d0ad187d6cff2c4e7fc4df96ed12948e0493ff687eb92; parent_hash: 0x4805…e1fa; extrinsics (2): [0xa8cf…0175, 0x0160…8389]]    
2023-05-23 14:15:48 🔖 Pre-sealed block for proposal at 1. Hash now 0x39f20cab22d26b2877a5b1d8dcf78a66d1bca0d84a6df1c465c58f8f0dff8a06, previously 0xb4fbb7932aaa80a0b08d0ad187d6cff2c4e7fc4df96ed12948e0493ff687eb92.    
2023-05-23 14:15:48 ✨ Imported #1 (0x39f2…8a06)    
2023-05-23 14:15:51 💤 Idle (0 peers), best: #1 (0x39f2…8a06), finalized #1 (0x39f2…8a06), ⬇ 0 ⬆ 0    
2023-05-23 14:15:56 💤 Idle (0 peers), best: #1 (0x39f2…8a06), finalized #1 (0x39f2…8a06), ⬇ 0 ⬆ 0    
2023-05-23 14:16:00 🙌 Starting consensus session on top of parent 0x39f20cab22d26b2877a5b1d8dcf78a66d1bca0d84a6df1c465c58f8f0dff8a06    
2023-05-23 14:16:00 🎁 Prepared block for proposing at 2 (0 ms) [hash: 0xc6296e9b2c71bd9b430b01eb15a3ecf2a740bf8e91cc3ecae565f1caefc074a6; parent_hash: 0x39f2…8a06; extrinsics (2): [0x5e09…d7b3, 0x9beb…7401]]    
2023-05-23 14:16:00 🔖 Pre-sealed block for proposal at 2. Hash now 0x11ab42932f766a552c3ee88c7d754e90513cf5ef0ee4d70bfaed11e569ee7c4a, previously 0xc6296e9b2c71bd9b430b01eb15a3ecf2a740bf8e91cc3ecae565f1caefc074a6.    
2023-05-23 14:16:00 ✨ Imported #2 (0x11ab…7c4a)    
2023-05-23 14:16:01 💤 Idle (0 peers), best: #2 (0x11ab…7c4a), finalized #2 (0x11ab…7c4a), ⬇ 0 ⬆ 0    
2023-05-23 14:16:06 💤 Idle (0 peers), best: #2 (0x11ab…7c4a), finalized #2 (0x11ab…7c4a), ⬇ 0 ⬆ 0    
2023-05-23 14:16:11 💤 Idle (0 peers), best: #2 (0x11ab…7c4a), finalized #2 (0x11ab…7c4a), ⬇ 0 ⬆ 0   
```

File: Cargo.lock [hunk 1]
```diff
@@ -7671,25 +7645,25 @@ dependencies = [
...
+source = "git+https://github.com/paritytech/substrate?branch=polkadot-v0.9.40#98f2e3451c9143278ec53c6718940aeabcd3b68a"
 dependencies = [
  "frame-benchmarking",
  "frame-support",
  "frame-system",
  "log",
  "parity-scale-codec",
  "scale-info",
- "sp-core",
+ "sp-core 7.0.0 (git+https://github.com/paritytech/substrate?branch=polkadot-v0.9.40)",
```
Comment (boundless-forest) [inline Cargo.lock:7656]: Why there are two kinds of `sp-core`?

```toml
[[package]]
name = "sp-core"
version = "7.0.0"
source = "git+https://github.com/purestake/substrate?branch=moonbeam-polkadot-v0.9.37#e66537fcb5bfbf804202b397195fd4fc23db008e"
dependencies = [
 "array-bytes 4.2.0",
 "base58",
 "bitflags",
 "blake2",
 "dyn-clonable",
 "ed25519-zebra",
 "futures 0.3.28",
 "hash-db 0.15.2",
 "hash256-std-hasher",
 "impl-serde",
 "lazy_static",
 "libsecp256k1",
 "log",
 "merlin",
 "parity-scale-codec",
 "parking_lot 0.12.1",
 "primitive-types 0.12.1",
 "rand 0.8.5",
 "regex",
 "scale-info",
 "schnorrkel",
 "secp256k1",
 "secrecy",
 "serde",
 "sp-core-hashing 5.0.0 (git+https://github.com/purestake/substrate?branch=moonbeam-polkadot-v0.9.37)",
 "sp-debug-derive 5.0.0 (git+https://github.com/purestake/substrate?branch=moonbeam-polkadot-v0.9.37)",
 "sp-externalities 0.13.0 (git+https://github.com/purestake/substrate?branch=moonbeam-polkadot-v0.9.37)",
 "sp-runtime-interface 7.0.0 (git+https://github.com/purestake/substrate?branch=moonbeam-polkadot-v0.9.37)",
 "sp-std 5.0.0 (git+https://github.com/purestake/substrate?branch=moonbeam-polkadot-v0.9.37)",
 "sp-storage 7.0.0 (git+https://github.com/purestake/substrate?branch=moonbeam-polkadot-v0.9.37)",
 "ss58-registry",
 "substrate-bip39",
 "thiserror",
 "tiny-bip39",
 "zeroize",
]

[[package]]
name = "sp-core"
version = "7.0.0"
source = "git+https://github.com/paritytech/substrate?branch=polkadot-v0.9.40#98f2e3451c9143278ec53c6718940aeabcd3b68a"
dependencies = [
 "array-bytes 4.2.0",
 "base58",
 "bitflags",
 "blake2",
 "bounded-collections",
 "dyn-clonable",
 "ed25519-zebra",
 "futures 0.3.28",
 "hash-db 0.16.0",
 "hash256-std-hasher",
 "impl-serde",
 "lazy_static",
 "libsecp256k1",
 "log",
 "merlin",
 "parity-scale-codec",
 "parking_lot 0.12.1",
 "primitive-types 0.12.1",
 "rand 0.8.5",
 "regex",
 "scale-info",
 "schnorrkel",
 "secp256k1",
 "secrecy",
 "serde",
 "sp-core-hashing 5.0.0 (git+https://github.com/paritytech/substrate?branch=polkadot-v0.9.40)",
 "sp-debug-derive 5.0.0 (git+https://github.com/paritytech/substrate?branch=polkadot-v0.9.40)",
 "sp-externalities 0.13.0 (git+https://github.com/paritytech/substrate?branch=polkadot-v0.9.40)",
 "sp-runtime-interface 7.0.0 (git+https://github.com/paritytech/substrate?branch=polkadot-v0.9.40)",
 "sp-std 5.0.0 (git+https://github.com/paritytech/substrate?branch=polkadot-v0.9.40)",
 "sp-storage 7.0.0 (git+https://github.com/paritytech/substrate?branch=polkadot-v0.9.40)",
 "ss58-registry",
 "substrate-bip39",
 "thiserror",
 "tiny-bip39",
 "zeroize",
]
```

File: Cargo.lock [hunk 2]
```diff
@@ -7734,56 +7704,56 @@ dependencies = [
...
- "pallet-timestamp",
  "parity-scale-codec",
  "rlp",
  "scale-info",
- "sp-core",
+ "sp-core 7.0.0 (git+https://github.com/paritytech/substrate?branch=polkadot-v0.9.40)",
  "sp-io",
  "sp-runtime",
- "sp-std",
+ "sp-std 5.0.0 (git+https://github.com/paritytech/substrate?branch=polkadot-v0.9.40)",
```
Comment (boundless-forest) [inline Cargo.lock:7732]: The same as `sp-core`, two version.

File: core/primitives/Cargo.toml [hunk 3]
```diff
@@ -11,7 +11,7 @@ version.workspace = true
 dc-types = { workspace = true }
 
 # moonbeam
```
Comment (boundless-forest) [inline core/primitives/Cargo.toml:13]: Please update this comment to `frontier`.

File: node/src/command.rs [hunk 4]
```diff
@@ -595,48 +595,51 @@ pub fn run() -> Result<()> {
...
 				sp_io::SubstrateHostFunctions,
 				<E as NativeExecutionDispatch>::ExtendHostFunctions,
 			>;
 
+
 			// grab the task manager.
 			let registry = &runner.config().prometheus_config.as_ref().map(|cfg| &cfg.registry);
 			let task_manager = TaskManager::new(runner.config().tokio_handle.clone(), *registry)
 				.map_err(|e| format!("Error: {:?}", e))?;
+			let info_provider = block_building_info::timestamp_with_aura_info(6000);
```
Comment (boundless-forest) [inline node/src/command.rs:616]: Find https://github.com/paritytech/cumulus/pull/2213, but still curious about the `6000` here, the blocktime is 12s normally for parachain, why is the blocktime here is 6s?  Do we need to dig deeper?

Comment (AurevoirXavier) [reply node/src/command.rs:616]: I think it's okay. It's only used for try runtime to build the block.

File: runtime/common/src/gov_origin.rs [hunk 5]
```diff
@@ -19,7 +19,7 @@
 // darwinia
 use dc_primitives::AccountId;
 // moonbeam
```
[the remark this answers is not in the document]
Comment (AurevoirXavier) [reply runtime/common/src/gov_origin.rs:21]: #1155

**Pull request merged**